packages/core/src/utils/styleUtils/stylesUtils.ts

```diff
@@ -20,19 +26,22 @@ export const toCSSAttribute = (key: string) => {
   return val;
 };
 
-export const buildStyleTag = ({ styles, nodeId }: { styles: CSSProperties; nodeId?: string }) => {
-  const stylesStr = Object.entries(styles)
+export const createJoinedCSSRules = (
+  cssProperties: CSSProperties,
+  useWhitespaces: boolean = false,
+) => {
+  const rules = Object.entries(cssProperties)
     .filter(([, value]) => value !== undefined)
-    .reduce(
-      (acc, [key, value]) =>
-        `${acc}
-        ${toCSSAttribute(key)}: ${value};`,
-      '',
+    .map(([key, value]) =>
+      useWhitespaces ? `${toCSSAttribute(key)}: ${value};` : `${toCSSAttribute(key)}:${value};`,
     );
+  return rules.join(useWhitespaces ? '\n' : '');
+};
```

Comment:
do we have a use case where we need white spaces?

Reply:
I didn't want to change the existing logic that would use whitespace and newlines for the editor mode. I guess this was done for easier inspecting the CSS code.